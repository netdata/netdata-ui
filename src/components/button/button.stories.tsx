import React from "react"
import { storiesOf } from "@storybook/react"
import styled from "styled-components"

import { action } from "@storybook/addon-actions"
import { text, boolean, select } from "@storybook/addon-knobs"
import { readmeCleanup } from "../../../utils/readme"
import { Button } from "."
// @ts-ignore
import readme from "./README.md"
import { iconsList } from "../icon"

const buttonStory = storiesOf("COMPONENTS|Controls/Button", module)
const icons: string[] = Object.keys(iconsList)

const subData = {
  readme: {
    sidebar: readmeCleanup(readme),
  },
  jest: ["button.test.tsx"],
}

buttonStory.add(
  "Button",
  () => (
    <Button
      label={text("Label", "LABEL")}
      icon={select("Name", ["No icon", ...icons], "plus")}
      disabled={boolean("Disabled", false)}
      onClick={action("clicked")}
      flavor={select("Name", ["default", "hollow", "borderless"], "default")}
      danger={boolean("Danger", false)}
      warning={boolean("Warning", false)}
      small={boolean("Small (works when icon only)", false)}
      isLoading={boolean("Is loading", false)}
      loadingLabel={text("Loading label prop", "LOADING...")}
    />
  ),
  subData
)

const OverridedButton = styled(Button)`
  background-color: black;
  color: purple;
`

buttonStory.add(
  "Button with CSS override",
  () => (
    <OverridedButton
      label={text("Label", "LABEL")}
      icon={select("Name", ["No icon", ...icons], "plus")}
      disabled={boolean("Disabled", false)}
      onClick={action("clicked")}
      flavor={select("Name", ["default", "hollow", "borderless"], "default")}
      danger={boolean("Danger", false)}
      warning={boolean("Warning", false)}
      small={boolean("Small (works when icon only)", false)}
      isLoading={boolean("Is loading", false)}
      loadingLabel={text("Loading label prop", "LOADING...")}
    />
  ),
  subData
)
