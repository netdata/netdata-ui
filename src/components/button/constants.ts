export const DEFAULT = "default"
export const HOLLOW = "hollow"
export const BORDER_LESS = "borderless"
export const BUTTON_BORDER = 3
