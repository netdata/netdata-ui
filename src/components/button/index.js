export { Button } from "./button"
export { default as IconButton } from "./IconButton"
export { ButtonGroup } from "./buttonGroup"
