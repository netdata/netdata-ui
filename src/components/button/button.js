import React, { forwardRef } from "react"
import { StyledButton } from "./styled"
import { Icon } from "@/components/icon"
import { LoaderIcon } from "@/components/icon/components"
import Flex from "@/components/templates/flex"

export const Button = forwardRef(
  (
    {
      label,
      icon,
      flavour,
      isLoading,
      loadingLabel,
      onClick,
      textTransform = "firstLetter",
      iconColor,
      iconSize,
      iconWidth,
      iconHeight,
      children = label,
      ...rest
    },
    ref
  ) => (
    <StyledButton
      flavour={flavour}
      textTransform={textTransform}
      hasIcon={!!icon || isLoading}
      onClick={isLoading ? undefined : onClick}
      ref={ref}
      iconColor={iconColor}
      {...rest}
    >
      {isLoading && <LoaderIcon className="button-icon" />}
      {icon && !isLoading && (
        <Flex justifyContent="center" alignItems="center" width="auto" height="100%">
          <Icon
            size={iconSize}
            className={iconColor ? "button-icon__color" : "button-icon"}
            title={icon}
            name={icon}
            width={iconWidth}
            height={iconHeight}
          />
        </Flex>
      )}

      {!!children && <span>{(isLoading && loadingLabel) || children}</span>}
    </StyledButton>
  )
)

Button.defaultProps = {
  onClick: () => {},
  icon: null,
}
