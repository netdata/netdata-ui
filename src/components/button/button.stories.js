import React, { useState } from "react"
import { Button, IconButton, ButtonGroup } from "."
import { iconsList } from "@/components/icon"

const icons = Object.keys(iconsList)

export const Basic = args => <Button {...args} />

export const BaseIconButton = args => (
  <IconButton
    {...args}
    label=""
    cursor="pointer"
    icon="chevron_left"
    iconSize="small"
    tooltip="Previous"
  />
)

const radioButtonItems = [
  { label: "One", value: 1 },
  { label: "Two", value: 2 },
  { label: "Three", value: 3 },
]

export const RadioButtonGroup = args => {
  const [checked, setChecked] = useState(1)
  const onChange = value => setChecked(value)

  return (
    <ButtonGroup
      items={radioButtonItems.map(item => ({ ...args, ...item }))}
      checked={checked}
      onChange={onChange}
    />
  )
}

export default {
  component: Button,
  tags: ["autodocs"],
  args: {
    onClick: () => alert("clicked"),
    flavour: "default",
    id: "some-id",
    className: "some-classname",
    label: "My button",
    loadingLabel: "Loading button",
    icon: icons[0],
    isLoading: false,
    disabled: false,
    danger: false,
    warning: false,
    small: false,
    neutral: false,
  },
  argTypes: {
    flavour: {
      options: ["default", "hollow", "borderless"],
      control: { type: "radio" },
    },
    id: { control: "text" },
    className: { control: "text" },
    label: { control: "text" },
    loadingLabel: { control: "text" },
    icon: { options: icons, type: "select" },
    isLoading: { control: "boolean" },
    disabled: { control: "boolean" },
    danger: { control: "boolean" },
    warning: { control: "boolean" },
    small: { control: "boolean" },
    neutral: { control: "boolean" },
  },
}
