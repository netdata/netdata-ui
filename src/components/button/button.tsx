import React, { SyntheticEvent, FC } from "react"
import { StyledButton } from "./styled"
import { Icon } from "../icon"
import { LoaderIcon } from "../icon/components"

export type ButtonType = "default" | "hollow" | "borderless"

export interface ButtonProps {
  id?: string
  className?: string
  onClick?: (e: SyntheticEvent<HTMLButtonElement>) => void
  danger?: boolean
  warning?: boolean
  type?: ButtonType | undefined
  flavor?: ButtonType
  disabled?: boolean
  small?: boolean
}

export interface ButtonWrapperProps extends ButtonProps {
  label?: string | JSX.Element
  icon?: any
  isLoading?: boolean
  loadingLabel?: string | JSX.Element
  loadingIcon?: any
}

export const Button: FC<ButtonWrapperProps> = ({
  label,
  icon,
  type,
  flavor,
  isLoading,
  loadingLabel,
  loadingIcon,
  ...rest
}: ButtonWrapperProps) => (
  <StyledButton flavor={type || flavor} hasLabel={!!label} {...rest}>
    {isLoading && !loadingIcon && !loadingIcon && <LoaderIcon className="button-icon" />}
    {icon && !isLoading && !loadingIcon && (
      <Icon className="button-icon" title={isLoading ? loadingIcon : icon} name={icon} />
    )}
    {label && <span>{(isLoading && loadingLabel) || label}</span>}
  </StyledButton>
)

Button.defaultProps = {
  onClick: () => {},
  icon: null,
  loadingIcon: null,
}
