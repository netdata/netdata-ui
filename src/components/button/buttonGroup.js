import React, { Children, isValidElement, cloneElement } from "react"
import Flex from "@/components/templates/flex"
import { Button } from "./button"

const getButtonGroupProps = (itemIndex, itemsLength) => {
  const groupFirst = itemIndex == 0
  const groupLast = itemIndex == itemsLength - 1

  return {
    groupFirst,
    groupLast,
    groupMiddle: !groupFirst && !groupLast,
  }
}

const Content = ({ children }) => {
  const renderedChildren = Children.toArray(children).length

  return (
    <>
      {Children.map(children, (child, index) => {
        if (isValidElement(child)) {
          const buttonGroupProps = getButtonGroupProps(index, renderedChildren)
          return cloneElement(child, buttonGroupProps)
        }

        return child
      })}
    </>
  )
}

const RadioButtons = ({ items, checked, buttonProps = {}, onChange }) => (
  <>
    {items.map(({ label, value }, index) => {
      const buttonGroupProps = getButtonGroupProps(index, items.length)
      return (
        <Button
          key={value}
          label={label}
          onClick={() => onChange(value)}
          {...(checked != value ? { flavour: "hollow" } : {})}
          {...buttonGroupProps}
          {...buttonProps}
        />
      )
    })}
  </>
)

export const ButtonGroup = ({ items, checked, onChange, children, buttonProps, ...props }) => (
  <Flex alignItems="center" {...props}>
    {items?.length ? (
      <RadioButtons items={items} checked={checked} onChange={onChange} buttonProps={buttonProps} />
    ) : (
      <Content>{children}</Content>
    )}
  </Flex>
)
