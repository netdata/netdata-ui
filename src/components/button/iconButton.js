import React, { forwardRef } from "react"
import Box from "@/components/templates/box"
import Flex from "@/components/templates/flex"
import Tooltip from "@/components/drops/tooltip"
import { TextSmall } from "@/components/typography"
import { Button } from "./button"

//this  addition here has been done to be aligned with our current implementation at cloud-frontend src/components/tooltips/customTooltip.js

const tooltipBackground = ["neutral", "black"]

const CustomTooltipContent = ({ content }) => (
  <Flex padding={[1.5, 2]} margin={[2]} background={tooltipBackground} round={1} alignSelf="start">
    <TextSmall color="bright">{content}</TextSmall>
  </Flex>
)

const IconButton = forwardRef(
  (
    {
      iconColor = "nodeBadgeColor",
      flavour = "borderless",
      icon,
      disabled,
      onClick,
      width = "20px",
      height = "20px",
      iconSize,
      tooltip = "",
      ...props
    },
    ref
  ) => {
    const isDefaultFlavour = flavour === "default"
    return (
      <Tooltip plain animation content={tooltip && <CustomTooltipContent content={tooltip} />}>
        <Box
          cursor="pointer"
          iconWidth={width}
          iconHeight={height}
          onClick={onClick}
          as={Button}
          flavour={flavour}
          disabled={disabled}
          icon={icon}
          iconColor={isDefaultFlavour ? "white" : iconColor}
          iconSize={iconSize}
          neutral={!isDefaultFlavour}
          ref={ref}
          {...props}
        />
      </Tooltip>
    )
  }
)

export default IconButton
