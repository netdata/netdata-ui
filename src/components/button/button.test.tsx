/**
 * @jest-environment jsdom
 */

import "@testing-library/jest-dom/extend-expect"
import "jest-styled-components"
import { fireEvent } from "@testing-library/react"
import { Button, ButtonWrapperProps } from "./button"
import { DefaultTheme } from "../../theme/default"
import { testWrapper } from "../../../test-utils"

describe("Button states", () => {
  it(" * should render disabled", () => {
    const { container } = testWrapper<ButtonWrapperProps>(
      Button,
      { label: "Test prop text", disabled: true },
      DefaultTheme,
      null
    )
    const button = container.firstChild
    expect(button).toHaveStyleRule("opacity", "0.4")
    expect(button).toHaveStyleRule("pointer-events", "none")
  })

  it(" * should render loading text", () => {
    const { getByText } = testWrapper<ButtonWrapperProps>(
      Button,
      { label: "Test prop text", loadingLabel: "loading...", isLoading: true },
      DefaultTheme,
      null
    )
    expect(getByText(/loading/)).toBeInTheDocument()
  })

  it(" * should render with icon", () => {
    const { getByText, getByTitle } = testWrapper<ButtonWrapperProps>(
      Button,
      { label: "Test prop text", icon: "plus" },
      DefaultTheme,
      null
    )

    expect(getByText(/prop text/)).toBeInTheDocument()
    expect(getByTitle("plus")).toBeInTheDocument()
  })

  it(" * should render only icon", () => {
    const { container } = testWrapper<ButtonWrapperProps>(
      Button,
      { type: "hollow", icon: "plus" },
      DefaultTheme,
      null
    )
    const button = container.firstChild
    expect(button).toMatchSnapshot()
    expect(button).toHaveStyleRule("width", "32px")
    expect(button).toHaveStyleRule("height", "32px")
  })

  it(" * should render smaller only icon", () => {
    const { container } = testWrapper<ButtonWrapperProps>(
      Button,
      { type: "hollow", icon: "plus", small: true },
      DefaultTheme,
      null
    )
    const button = container.firstChild
    expect(button).toMatchSnapshot()
    expect(button).toHaveStyleRule("width", "24px")
    expect(button).toHaveStyleRule("height", "24px")
  })

  it(" * should render loading icon", () => {
    const { container, getByText, queryByTitle } = testWrapper<ButtonWrapperProps>(
      Button,
      { label: "Test prop text", icon: "plus", isLoading: true },
      DefaultTheme,
      null
    )

    expect(container.firstChild).toMatchSnapshot()
    expect(getByText(/prop text/)).toBeInTheDocument()
    expect(queryByTitle("plus")).not.toBeInTheDocument()
  })

  it(" * should be clickable", () => {
    const onClickFn = jest.fn()
    const { getByText } = testWrapper<ButtonWrapperProps>(
      Button,
      { label: "Test prop text", onClick: onClickFn },
      DefaultTheme,
      null
    )

    fireEvent.click(getByText(/prop text/))
    expect(onClickFn).toBeCalled()
  })
})

describe("Default Button", () => {
  it(" * should render", () => {
    const { container, getByText } = testWrapper<ButtonWrapperProps>(
      Button,
      { label: "Test prop text" },
      DefaultTheme,
      null
    )

    expect(getByText(/prop text/)).toBeInTheDocument()

    const button = container.firstChild
    expect(button).toMatchSnapshot()
    expect(button).toHaveStyleRule("width", "128px")
    expect(button).toHaveStyleRule("height", "40px")
    expect(button).toHaveStyleRule("background-color", "#00AB44")
    expect(button).toHaveStyleRule("border-color", "#00AB44")
    expect(button).toHaveStyleRule("color", "#FFF")
    expect(button).toHaveStyleRule("opacity", "1")
    expect(button).toHaveStyleRule("pointer-events", "auto")
  })

  it(" * should render for danger", () => {
    const { container } = testWrapper<ButtonWrapperProps>(
      Button,
      { label: "Test prop text", danger: true },
      DefaultTheme,
      null
    )
    const button = container.firstChild
    expect(button).toHaveStyleRule("background-color", "#FF4136")
    expect(button).toHaveStyleRule("border-color", "#FF4136")
    expect(button).toHaveStyleRule("color", "#FFF")
  })

  it(" * should render for warning", () => {
    const { container } = testWrapper<ButtonWrapperProps>(
      Button,
      { label: "Test prop text", warning: true },
      DefaultTheme,
      null
    )
    const button = container.firstChild
    expect(button).toHaveStyleRule("background-color", "#FFC300")
    expect(button).toHaveStyleRule("border-color", "#FFC300")
    expect(button).toHaveStyleRule("color", "#FFF")
  })
})

describe("Hollow Button", () => {
  it(" * should render", () => {
    const { container, getByText } = testWrapper<ButtonWrapperProps>(
      Button,
      { label: "Test prop text", type: "hollow" },
      DefaultTheme,
      null
    )

    expect(getByText(/prop text/)).toBeInTheDocument()

    const button = container.firstChild
    expect(button).toMatchSnapshot()
    expect(button).toHaveStyleRule("width", "128px")
    expect(button).toHaveStyleRule("height", "40px")
    expect(button).toHaveStyleRule("background-color", "#FFF")
    expect(button).toHaveStyleRule("border-color", "#00AB44")
    expect(button).toHaveStyleRule("color", "#00AB44")
    expect(button).toHaveStyleRule("opacity", "1")
    expect(button).toHaveStyleRule("pointer-events", "auto")
  })

  it(" * should render for danger", () => {
    const { container } = testWrapper<ButtonWrapperProps>(
      Button,
      { label: "Test prop text", type: "hollow", danger: true },
      DefaultTheme,
      null
    )
    const button = container.firstChild
    expect(button).toHaveStyleRule("background-color", "#FFF")
    expect(button).toHaveStyleRule("border-color", "#FF4136")
    expect(button).toHaveStyleRule("color", "#FF4136")
  })

  it(" * should render for warning", () => {
    const { container } = testWrapper<ButtonWrapperProps>(
      Button,
      { label: "Test prop text", type: "hollow", warning: true },
      DefaultTheme,
      null
    )
    const button = container.firstChild
    expect(button).toHaveStyleRule("background-color", "#FFF")
    expect(button).toHaveStyleRule("border-color", "#FFC300")
    expect(button).toHaveStyleRule("color", "#FFC300")
  })
})

describe("Borderless Button", () => {
  it(" * should render", () => {
    const { container, getByText } = testWrapper<ButtonWrapperProps>(
      Button,
      { label: "Test prop text", type: "borderless" },
      DefaultTheme,
      null
    )

    expect(getByText(/prop text/)).toBeInTheDocument()

    const button = container.firstChild
    expect(button).toMatchSnapshot()
    expect(button).toHaveStyleRule("width", "128px")
    expect(button).toHaveStyleRule("height", "40px")
    expect(button).toHaveStyleRule("background-color", "#FFF")
    expect(button).toHaveStyleRule("border-color", "#FFF")
    expect(button).toHaveStyleRule("color", "#00AB44")
    expect(button).toHaveStyleRule("opacity", "1")
    expect(button).toHaveStyleRule("pointer-events", "auto")
  })

  it(" * should render for danger", () => {
    const { container } = testWrapper<ButtonWrapperProps>(
      Button,
      { label: "Test prop text", type: "borderless", danger: true },
      DefaultTheme,
      null
    )
    const button = container.firstChild
    expect(button).toHaveStyleRule("background-color", "#FFF")
    expect(button).toHaveStyleRule("border-color", "#FFF")
    expect(button).toHaveStyleRule("color", "#FF4136")
  })

  it(" * should render for warning", () => {
    const { container } = testWrapper<ButtonWrapperProps>(
      Button,
      { label: "Test prop text", type: "borderless", warning: true },
      DefaultTheme,
      null
    )
    const button = container.firstChild
    expect(button).toHaveStyleRule("background-color", "#FFF")
    expect(button).toHaveStyleRule("border-color", "#FFF")
    expect(button).toHaveStyleRule("color", "#FFC300")
  })
})
