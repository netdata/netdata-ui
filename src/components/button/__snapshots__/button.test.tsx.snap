// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Borderless Button  * should render 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  width: 128px;
  height: 40px;
  font-weight: bold;
  font-size: 12px;
  line-height: 16px;
  white-space: nowrap;
  word-break: keep-all;
  cursor: pointer;
  opacity: 1;
  pointer-events: auto;
  padding: 8px;
  -webkit-transition: all 150ms;
  transition: all 150ms;
  background-color: #FFF;
  color: #00AB44;
  border-width: 1px;
  border-style: solid;
  border-color: #FFF;
  border-radius: 2px;
  box-sizing: border-box;
}

.c0:hover {
  border-color: #FFF;
  background-color: #FFF;
  color: #00CB51;
}

.c0:hover .button-icon {
  fill: #00CB51;
}

.c0:active {
  border-color: #FFF;
  background-color: #FFF;
  color: #00AB44;
}

.c0:focus {
  outline: none;
}

.c0 .button-icon {
  position: absolute;
  left: 4px;
  height: 16px;
  width: 16px;
  fill: #00AB44;
}

.c0 .ntd-spinner {
  fill: none;
  stroke-width: 17px;
  stroke-dasharray: 100;
  stroke-dashoffset: 100;
  -webkit-animation: ntd-draw 1s linear infinite;
  animation: ntd-draw 1s linear infinite;
  stroke: #fff;
  width: 24px;
}

.c0 .path {
  stroke: #fff;
}

<button
  class="c0"
>
  <span>
    Test prop text
  </span>
</button>
`;

exports[`Button states  * should render loading icon 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  width: 128px;
  height: 40px;
  font-weight: bold;
  font-size: 12px;
  line-height: 16px;
  white-space: nowrap;
  word-break: keep-all;
  cursor: pointer;
  opacity: 1;
  pointer-events: auto;
  padding: 8px;
  -webkit-transition: all 150ms;
  transition: all 150ms;
  background-color: #00AB44;
  color: #FFF;
  border-width: 1px;
  border-style: solid;
  border-color: #00AB44;
  border-radius: 2px;
  box-sizing: border-box;
}

.c0:hover {
  border-color: #00CB51;
  background-color: #00AB44;
  color: #FFF;
}

.c0:hover .button-icon {
  fill: #FFF;
}

.c0:active {
  border-color: #00CB51;
  background-color: #00CB51;
  color: #FFF;
}

.c0:focus {
  outline: none;
}

.c0 .button-icon {
  position: absolute;
  left: 4px;
  height: 16px;
  width: 16px;
  fill: #FFF;
}

.c0 .ntd-spinner {
  fill: none;
  stroke-width: 17px;
  stroke-dasharray: 100;
  stroke-dashoffset: 100;
  -webkit-animation: ntd-draw 1s linear infinite;
  animation: ntd-draw 1s linear infinite;
  stroke: #fff;
  width: 24px;
}

.c0 .path {
  stroke: #fff;
}

.c1 {
  fill: none;
  stroke-width: 17px;
  stroke-dasharray: 100;
  stroke-dashoffset: 100;
  -webkit-animation: ntd-draw 1s linear infinite;
  animation: ntd-draw 1s linear infinite;
  stroke: #fff;
  width: 24px;
}

.c1 .path {
  stroke: #fff;
}

<button
  class="c0"
>
  <svg
    class="c1 button-icon"
    version="1.1"
    viewBox="0 0 21 17"
    xmlns="http://www.w3.org/2000/svg"
  >
    <g
      class="path"
      fill="none"
      fill-rule="evenodd"
      stroke="none"
      stroke-width="1"
    >
      <path
        d="M2,1 C8.25086152,1 11.9367136,1 13.0575562,1 C14.73882,1 19.6834591,2 19.9614325,7.72050108 C20.239406,13.4410022 15.7459591,15.1224845 13.6463763,15.1224845 C12.2466545,15.1224845 10.0279195,15.1224845 6.9901715,15.1224845 L2,1 Z"
        id="Path-2"
        stroke-width="2"
      />
    </g>
  </svg>
  <span>
    Test prop text
  </span>
</button>
`;

exports[`Button states  * should render only icon 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  width: 32px;
  height: 32px;
  font-weight: bold;
  font-size: 12px;
  line-height: 16px;
  white-space: nowrap;
  word-break: keep-all;
  cursor: pointer;
  opacity: 1;
  pointer-events: auto;
  padding: 8px;
  -webkit-transition: all 150ms;
  transition: all 150ms;
  background-color: #FFF;
  color: #00AB44;
  border-width: 1px;
  border-style: solid;
  border-color: #00AB44;
  border-radius: 2px;
  box-sizing: border-box;
}

.c0:hover {
  border-color: #00CB51;
  background-color: #FFF;
  color: #00CB51;
}

.c0:hover .button-icon {
  fill: #00CB51;
}

.c0:active {
  border-color: #00CB51;
  background-color: #00CB51;
  color: #FFF;
}

.c0:focus {
  outline: none;
}

.c0 .button-icon {
  position: absolute;
  left: auto;
  height: 16px;
  width: 16px;
  fill: #00AB44;
}

.c0 .ntd-spinner {
  fill: none;
  stroke-width: 17px;
  stroke-dasharray: 100;
  stroke-dashoffset: 100;
  -webkit-animation: ntd-draw 1s linear infinite;
  animation: ntd-draw 1s linear infinite;
  stroke: #fff;
  width: 24px;
}

.c0 .path {
  stroke: #fff;
}

.c1 {
  height: 24px;
  width: 24px;
}

<button
  class="c0"
>
  <svg
    class="c1 button-icon"
    title="plus"
  >
    <use
      xlink:href="#undefined"
    />
  </svg>
</button>
`;

exports[`Button states  * should render smaller only icon 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  width: 24px;
  height: 24px;
  font-weight: bold;
  font-size: 12px;
  line-height: 16px;
  white-space: nowrap;
  word-break: keep-all;
  cursor: pointer;
  opacity: 1;
  pointer-events: auto;
  padding: 8px;
  -webkit-transition: all 150ms;
  transition: all 150ms;
  background-color: #FFF;
  color: #00AB44;
  border-width: 1px;
  border-style: solid;
  border-color: #00AB44;
  border-radius: 2px;
  box-sizing: border-box;
}

.c0:hover {
  border-color: #00CB51;
  background-color: #FFF;
  color: #00CB51;
}

.c0:hover .button-icon {
  fill: #00CB51;
}

.c0:active {
  border-color: #00CB51;
  background-color: #00CB51;
  color: #FFF;
}

.c0:focus {
  outline: none;
}

.c0 .button-icon {
  position: absolute;
  left: auto;
  height: 16px;
  width: 16px;
  fill: #00AB44;
}

.c0 .ntd-spinner {
  fill: none;
  stroke-width: 17px;
  stroke-dasharray: 100;
  stroke-dashoffset: 100;
  -webkit-animation: ntd-draw 1s linear infinite;
  animation: ntd-draw 1s linear infinite;
  stroke: #fff;
  width: 24px;
}

.c0 .path {
  stroke: #fff;
}

.c1 {
  height: 24px;
  width: 24px;
}

<button
  class="c0"
>
  <svg
    class="c1 button-icon"
    title="plus"
  >
    <use
      xlink:href="#undefined"
    />
  </svg>
</button>
`;

exports[`Default Button  * should render 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  width: 128px;
  height: 40px;
  font-weight: bold;
  font-size: 12px;
  line-height: 16px;
  white-space: nowrap;
  word-break: keep-all;
  cursor: pointer;
  opacity: 1;
  pointer-events: auto;
  padding: 8px;
  -webkit-transition: all 150ms;
  transition: all 150ms;
  background-color: #00AB44;
  color: #FFF;
  border-width: 1px;
  border-style: solid;
  border-color: #00AB44;
  border-radius: 2px;
  box-sizing: border-box;
}

.c0:hover {
  border-color: #00CB51;
  background-color: #00AB44;
  color: #FFF;
}

.c0:hover .button-icon {
  fill: #FFF;
}

.c0:active {
  border-color: #00CB51;
  background-color: #00CB51;
  color: #FFF;
}

.c0:focus {
  outline: none;
}

.c0 .button-icon {
  position: absolute;
  left: 4px;
  height: 16px;
  width: 16px;
  fill: #FFF;
}

.c0 .ntd-spinner {
  fill: none;
  stroke-width: 17px;
  stroke-dasharray: 100;
  stroke-dashoffset: 100;
  -webkit-animation: ntd-draw 1s linear infinite;
  animation: ntd-draw 1s linear infinite;
  stroke: #fff;
  width: 24px;
}

.c0 .path {
  stroke: #fff;
}

<button
  class="c0"
>
  <span>
    Test prop text
  </span>
</button>
`;

exports[`Hollow Button  * should render 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  width: 128px;
  height: 40px;
  font-weight: bold;
  font-size: 12px;
  line-height: 16px;
  white-space: nowrap;
  word-break: keep-all;
  cursor: pointer;
  opacity: 1;
  pointer-events: auto;
  padding: 8px;
  -webkit-transition: all 150ms;
  transition: all 150ms;
  background-color: #FFF;
  color: #00AB44;
  border-width: 1px;
  border-style: solid;
  border-color: #00AB44;
  border-radius: 2px;
  box-sizing: border-box;
}

.c0:hover {
  border-color: #00CB51;
  background-color: #FFF;
  color: #00CB51;
}

.c0:hover .button-icon {
  fill: #00CB51;
}

.c0:active {
  border-color: #00CB51;
  background-color: #00CB51;
  color: #FFF;
}

.c0:focus {
  outline: none;
}

.c0 .button-icon {
  position: absolute;
  left: 4px;
  height: 16px;
  width: 16px;
  fill: #00AB44;
}

.c0 .ntd-spinner {
  fill: none;
  stroke-width: 17px;
  stroke-dasharray: 100;
  stroke-dashoffset: 100;
  -webkit-animation: ntd-draw 1s linear infinite;
  animation: ntd-draw 1s linear infinite;
  stroke: #fff;
  width: 24px;
}

.c0 .path {
  stroke: #fff;
}

<button
  class="c0"
>
  <span>
    Test prop text
  </span>
</button>
`;
